File: Matury CKE/2022 Czerwiec/Programowanie/C++/zad4_1.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <vector>
#include <algorithm>

int reflection(std::string str)
{
    std::reverse(str.begin(), str.end());
    return std::stoi(str);
}

int main()
{
    std::cout << "4.1. ";

    // std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Czerwiec/DANE/przyklad.txt");
    std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Czerwiec/DANE/liczby.txt");

    std::vector<std::string> numsList;
    std::vector<int> reflectedAndDivBy17List;

    std::string num;
    while(dane >> num){
        numsList.push_back(num);
        int refNum = reflection(num);
        if(refNum % 17 == 0){
            reflectedAndDivBy17List.push_back(refNum);
            std::cout << refNum << " ";
        }
    }

    return 0;
}

File: Matury CKE/2022 Czerwiec/Programowanie/Python/zad4_1.py
# numbers=[i.strip() for i in open('../../DANE/przyklad.txt').readlines()]
numbers=[i.strip() for i in open('../../DANE/liczby.txt').readlines()]
reflections=[int(i[::-1]) for i in numbers]
reflections_divisible_by_17=[str(i) for i in reflections if i%17==0]

print(f'4.1. {" ".join(reflections_divisible_by_17)}')
